app/api/sessions.py

```diff
@@ -105,20 +109,20 @@ def query(self, view_kwargs):
         return query_
 
     def before_create_object(self, data, view_kwargs):
-        if view_kwargs.get('track_id') is not None:
-            track = self.session.query(Track).filter_by(id=view_kwargs['track_id']).one()
-            data['track_id'] = track.id
-        if view_kwargs.get('session_type_id') is not None:
-            session_type = self.session.query(SessionType).filter_by(
-                id=view_kwargs['session_type_id']).one()
-            data['session_type_id'] = session_type.id
-        if view_kwargs.get('microlocation_id') is not None:
-            microlocation = self.session.query(Microlocation).filter_by(
-                id=view_kwargs['microlocation_id']).one()
-            data['microlocation_id'] = microlocation.id
```

Comment:
One should be able to use identifier anywhere where event id is used. Public or private.